### PR TITLE
Fix authentication of edit and delete

### PR DESCRIPTION
**Pull request by mmezher**
After testing previous pull request, some features, such as deleting a poll, were still doable by users. Fixed constraints so that admin status is checked rather than managing polls. This also fixed the rspec admin test, which had been failing previously.

### CODE REVIEW

**Comment by gkop** [comment]
Rad! The hound complaints seem valid mostly, would you fix those and merge please?

File: app/controllers/polls_controller.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class PollsController < ApplicationController
-  
-  load_and_authorize_resource only: [:new, :create, :edit, :update]
+  load_and_authorize_resource only: [:new, :create, :edit, :update, 
+:destroy, :published]
```
Comment (mmezher) [inline app/controllers/polls_controller.rb]: Oops, missed one. Will fix that shortly

**Comment by mmezher** [comment]
Unsure why the admin test has again stopped passing. The rspec file and all other parts other than proper formatting stayed the same.

**Comment by mmezher** [comment]
Strange behavior. All users can destroy polls still, but publish button is disabled, edit does not work, new poll does not show up nor work via direct url, and users still see all buttons. However, non-users can no longer access polls viewing, so their access to destroying polls has been disabled. Still need to fix authorization of destruction. Will need to include another rspec test for users/admin deleting in next pull request.

**Pull request merged**